### PR TITLE
Enable partition awareness.

### PR DESCRIPTION
**Pull request by jeschkies**

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/MarathonSchedulerDriver.scala [hunk 1]
```diff
@@ -73,6 +73,9 @@ object MarathonSchedulerDriver {
       log.debug("GPU_RESOURCES feature enabled.")
     }
 
+    // Being partition aware allows Marathon to more granular task status, see MESOS-5344.
+    frameworkInfoBuilder.addCapabilities(Capability.newBuilder().setType(Capability.Type.PARTITION_AWARE))
```
Comment (jeschkies) [inline src/main/scala/mesosphere/marathon/MarathonSchedulerDriver.scala]: What happens if this is called with Mesos 1.0?

Comment (nfnt) [inline src/main/scala/mesosphere/marathon/MarathonSchedulerDriver.scala]: Nothing, the enum values are backwards-compatible.

File: task_status_fsm.dot [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    RUNNING -> FINISHED;
+    RUNNING -> UNKNOWN;
+    RUNNING -> UNREACHABLE;
+    RUNNING -> KILLING;
+
+    KILLING -> KILLED;
+
+    GONE_BY_OPERATOR -> GONE;
+    GONE_BY_OPERATOR -> RUNNING;
+}
```
Comment (jeschkies) [inline task_status_fsm.dot]: I'd like to take the chance to discuss the different stats and their transitions. So far I inferred everything from the `mesos.proto`. I'd like to add what Marathon should be doing once it's partition aware.

Comment (jeschkies) [inline task_status_fsm.dot]: Task State Machine

File: task_status_fsm.dot [hunk 3]
```diff
@@ -0,0 +1,49 @@
+digraph task_status_fsm {
```
Comment (jasongilanfarr) [inline task_status_fsm.dot]: :+1: :100: :+1: 

but can you put in in docs or something? or even next to the MarathonTaskStatus...

**Comment by jeschkies** [comment]
Task Status FSM

This is what I got mostly from @neilconway.

As far as I can tell, Marathon will have to handle cases when a task comes back from being `UNREACHABLE`.

**Comment by jeschkies** [comment]
Task Status FSM

**Comment by jeschkies** [comment]
Code review moved to https://phabricator.mesosphere.com/D39

**Comment by jeschkies** [comment]
See https://phabricator.mesosphere.com/D50 and follow ups.

**Pull request closed without merging**